Formula/m/mariadb@11.1.rb

```diff
@@ -0,0 +1,184 @@
+class MariadbAT111 < Formula
+  desc "Drop-in replacement for MySQL"
+  homepage "https://mariadb.org/"
+  url "https://archive.mariadb.org/mariadb-11.1.3/source/mariadb-11.1.3.tar.gz"
+  sha256 "ad3aa6ae478179275ecb28e3ed8a578f2004f084e14409727c0bf04975ace36c"
+  license "GPL-2.0-only"
+
+  livecheck do
+    url "https://downloads.mariadb.org/rest-api/mariadb/all-releases/?olderReleases=false"
+    strategy :json do |json|
+      json["releases"]&.map do |release|
+        next unless release["release_number"]&.start_with?(version.major_minor)
+        next if release["status"] != "stable"
+
+        release["release_number"]
+      end
+    end
+  end
+
+  keg_only :versioned_formula
+
+  # See: https://mariadb.com/kb/en/changes-improvements-in-mariadb-11-1/
+  # End-of-life on 2024-08-21: https://mariadb.org/about/#maintenance-policy
+  deprecate! date: "2024-08-21", because: :unsupported
```

Comment:
Let's disable it at that date then

Reply:
Should this be deleted for now, or is it preferable to comment the line?
BTW, almost all other versioned MariaDB formulae have similar lines with future deprecation dates.

Reply:
Let's ask @Homebrew/core. But if upstream drops support at that date I think it should be:
```suggestion
  disable! date: "2024-08-21", because: :unsupported
```

Which would deprecate it immediately, and disable it after that date.

Reply:
I'd rather deprecate rather than disable when it's unsupported. No reason to stop people installing it entirely if it still works for them and there's no known high-CVE issues.

Reply:
yeah, I agree, deprecate would be better unless there is critical CVE issues

Reply:
So we're moving away from our "only support what upstream supports"-policy to "if it's deprecated on our side we'll ship it regardless of the upstream state"-policy then?

Reply:
My understanding was we were only deprecating things that are unsupported at first and not disabling/removing them until later. I could be confusing what we were/are doing. Is the policy documented somewhere as that'd be easier to discuss than this specific line.

Reply:
My thought process is:

- If a project has a "security updates only" phase, it should be deprecated when full support ends and then disabled when security updates end.
- If a project doesn't have a "security updates only" phase, it should be deprecated for ~90 days, before being disabled.

The current "deprecate for a year, then disable for a year" seem like far too long for us to be shipping unsupported, potentially vulnerable software. New CVE's are often not reported for unsupported products, so that's not a solid indicator of if we should disable.

Reply:
> I think a PR to the documentation to document this would be welcome, so we can align on the strategy.

Agreed. To be explicit: I'm 👍🏻 on @SMillerDev's suggested change here but would like to get us more aligned/documented on docs.brew.sh about this.

Reply:
> If a project has a "security updates only" phase, it should be deprecated when full support ends and then disabled when security updates end.

The thing with this is sometimes this phase is several years, and several years of deprecation dilutes what deprecation means, and then encourage a "not a problem" attitude. There really should be a maximum cap here of maybe a few months (potentially escalating based on popularity thresholds).

Deprecation ultimately is a signal that this package is going to soon no longer be available. It's a deprecation from Homebrew, not a deprecation from upstream (though they often align).

Reply:
It's worth noting also that deprecation has big impacts on test-bot behaviour. We currently don't test deprecated formulae _at all_ unless it's modified directly in the PR. This means revision bumps are missed, and the formulae are left badly broken in some scenarios.

If we start deprecating before EOL (which is not a bad idea on its own), we should probably consider adjusting this test bot behaviour accordingly. Especially if we're doing super long deprecation periods.

Reply:
> The thing with this is sometimes this phase is several years, and several years of deprecation dilutes what deprecation means and encourages a "not a problem" attitude. There really should be a maximum cap here of maybe 3 months (potentially esclating that period based on popularity thresholds).

I don't think the chance users stop listening to deprecations if we are honest about them should be a reason not to be honest about them.

> It's a deprecation from Homebrew, not a deprecation from upstream (though they can align).

The only way they don't align is if upstream stops supporting Homebrew's platforms (we'll be early), or if upstream archives their repo (we'll be late).

> If we start deprecating before EOL (which is not a bad idea on its own), we should probably consider adjusting this test bot behaviour accordingly. Especially if we're doing super long deprecation periods. One idea could be to continue to test those with :unsupported as a deprecation reason (and still skip :does_not_build and :unmaintained).

That sounds reasonable to me.

Reply:
> I don't think the chance users stop listening to deprecations if we are honest about them should be a reason not to be honest about them.

We don't show dates in our deprecation warnings, so it's arguably more honest to be telling them within a few months of its disable date rather than a deprecation period that might be a few weeks, might be a few years. It's way too inconsistent right now. For `brew` deprecations, we have a defined period where deprecations in "X.Y.0" are disabled in "X.(Y + 1).0", which are typically 3 times a year.

We could probably control this on the `brew` side where we only start warning if the disable date is X months away.

Reply:
We do sort of show it on formulae.brew.sh (it's a hover message). I agree we can improve the warnings though. Especially when we know something will be disabled.